addons/l10n_pk/demo/account_demo.py

```diff
@@ -0,0 +1,303 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+from odoo import api, fields, models, Command
+
+
+class AccountChartTemplate(models.AbstractModel):
+    _inherit = "account.chart.template"
+
+    @api.model
+    def _get_demo_data(self, company=False):
+        if company.account_fiscal_country_id.code == "PK":
+            return {
+                'res.partner': self._get_l10n_pk_demo_data_partner(company),
+                'account.move': self._get_demo_data_move(company),
+                'res.config.settings': self._get_l10n_pk_demo_data_config_settings(company),
+            }
+        else:
+            return super()._get_demo_data(company)
+
+    @api.model
+    def _get_l10n_pk_demo_data_partner(self, company=False):
+        company = company or self.env.company
+        state_id = self.env.ref('l10n_pk.state_pk_is')
+        default_partner_dict = {
+            'company_id': company.id,
+            'country_id': self.env.ref('base.pk').id,
+            'is_company': True
+        }
+        return {
+            'res_partner_punjab': {
+                **default_partner_dict,
+                'name': 'Punjab Customer',
+                'street': '38/A Gulberg Main Boulevard',
+                'street2': 'Gulberg III, Lahore',
+                'city': 'Lahore',
+                'state_id': company.state_id.id,
+                'zip': '54000',
+                'vat': '4174942',
+                'phone': '+929876543210',
+                'email': 'p086adrh7w@guerrillamail.com',
+                'website': 'punjabpartner.odoo.com',
+            },
+            'res_partner_islamabad': {
+                **default_partner_dict,
+                'name': 'Islamabad Vendor',
+                'street': 'Office 12, 3rd Floor, Giga Mall',
+                'street2': 'Main GT Road, DHA Phase II',
+                'city': 'Taxila',
+                'state_id': state_id.id,
+                'zip': '40000',
+                'vat': False,
+                'is_company': False,
+                'phone': '+921234567890',
+                'email': 'adrh7wp086@guerrillamail.com',
+                'website': 'islamabadvendor.odoo.com',
+            },
+        }
+
+    @api.model
+    def _get_demo_data_move(self, company=False):
+        cid = company.id or self.env.company.id
```

Comment:
```suggestion
        cid = (company or self.env.company).id
```
default parameter for company is False, you won't get id in that case